UnchainedLauncherCore/src/API/RegisteredServer.cs

```diff
@@ -132,36 +132,41 @@ public Task WhenA2SOk()
         private async Task MaintainRegistration(CancellationToken token)
         {
             A2sInfo a2sRes = await GetServerState(token);
-            var res = await backend.RegisterServerAsync(localIp, new(serverInfo, a2sRes));
-            logger.Info($"Server '{this.serverInfo.Name}' registered with backend.");
+            var res = await backend.RegisterServerAsync(localIp, new(ServerInfo, a2sRes));
+            logger.Info($"Server '{this.ServerInfo.Name}' registered with backend.");
             this.IsRegistrationOk = true;
             double refreshBefore = res.RefreshBefore;
             RemoteInfo = res.Server;
-            Task updateDelay = Task.Delay(updateIntervalMillis, token);
-            while (true)
-            {
-                double secondsUntilExpiry = refreshBefore - DateTimeOffset.Now.ToUnixTimeSeconds();
-                // do a heartbeat with half the time remaining
-                int heartBeatAfterSeconds = (int)Math.Ceiling(secondsUntilExpiry/2);
-                Task heartBeatDelay = Task.Delay(1000 * heartBeatAfterSeconds, token);
-                var fin = await Task.WhenAny(heartBeatDelay, updateDelay, token.WhenCanceled());
```

Comment:
The updateDelay was appropriately scheduled outside the loop, and then only re-scheduled after running an update. The heartbeat was just scheduled every iteration and used an updated `refreshBefore` value.  This would cause a race condition, as `secondsUntilExpiry/2` approaches 0, but is still slightly greater than `updateDelay`. Then an update would trigger and the request duration would take too long and allow the heartbeat to expire